src/theory/quantifiers/trigger.h

```diff
@@ -33,59 +33,89 @@ namespace inst {
 class IMGenerator;
 class InstMatchGenerator;
 
-/** Information about a node used in a trigger. */
+/** Information about a node used in a trigger. 
+*
+* As an example, consider the trigger 
+*   { f( x ), g( y ), P( y, z ) } 
+* for quantified formula 
+*   forall xy. ( f( x ) != b => ( P( x, g( y ) ) V P( y, z ) ) )
+* This example is referenced for each of the functions below.
+*/
 class TriggerTermInfo {
 public:
   TriggerTermInfo() : d_reqPol(0){}
   ~TriggerTermInfo(){}
-  /** the free variables in the node */
+  /** The free variables in the node 
+  *
+  * For the trigger term info for f( x ) in the above example, d_fv = { x } 
+  * For the trigger term info for g( y ) in the above example, d_fv = { y } 
+  * For the trigger term info for P( y, z ) in the above example, d_fv = { y, z } 
+  */
   std::vector< Node > d_fv;
-  /** required polarity */
+  /** Required polarity:  1 for equality, -1 for disequality, 0 for none
+  *
+  * For the trigger term info for f( x ) in the above example, d_reqPol = -1
+  * For the trigger term info for g( y ) in the above example, d_reqPol = 0 
+  * For the trigger term info for P( y, z ) in the above example,  d_reqPol = 1
+  */
   int d_reqPol;
-  /** required polarity equal term
-  * d_reqPolEq is not null, then n appears in an equality with d_reqPolEq.
+  /** Required polarity equal term
+  *
+  * If d_reqPolEq is not null, 
   *   if d_reqPol = 1, then this trigger term must be matched to terms in the equivalence class of d_reqPolEq
   *   if d_reqPol = -1, then this trigger term must be matched to terms *not* in the equivalence class of d_reqPolEq
-  * This information is determined by analyzing the entailed equalities and disequalities of quantified formulas.
+  *
+  * This information is typically chosen by analyzing the entailed equalities and disequalities of quantified formulas.
+  * For the trigger term info for f( x ) in the above example, d_reqPolEq = b
+  * For the trigger term info for g( y ) in the above example, d_reqPolEq = Node::null()
+  * For the trigger term info for P( y, z ) in the above example,  d_reqPolEq = false
+  *
+  * In other words, it is only useful to match f( x ) to terms not in the equivalence class of b,
+  * and P( y, z ) to terms in the equivalence class of false, as such instances will always be 
+  * redundant/entailed in the current context.
```

Comment:
This comment (lines 73-75) is needed to understand the rest.  Maybe move it to the class comment.  Then the individual method comments are clearer.  Also, the phrase starting "as such instances" applies only to f(x), right?  Can you rewrite to make it clear?